Actions/Deploy/Deploy.ps1

```diff
@@ -10,6 +10,80 @@ Param(
+            $appJson = Get-AppJsonFromAppFile -appFile $app
+            $isPTE = ($appjson.idRanges.from -lt 100000 -and $appjson.idRanges.from -ge 50000)
+            $installedApp = $installedApps | Where-Object { $_.id -eq $appJson.id }
+            $needsUpgrade = $false
+            $needsInstall = $false
+            if ($installedApp) {
+                $newVersion = [version]::new($appJson.Version)
+                $installedVersion = [version]::new($installedApp.versionMajor, $installedApp.versionMinor, $installedApp.versionBuild, $installedApp.versionRevision)
+                if ($newVersion -gt $installedVersion) {
+                    $msg = "Dependency app $($appJson.name) is already installed in version $installedVersion, which is lower than $newVersion."
+                    if ($installMode -eq 'upgrade') {
+                        Write-Host "$msg Needs upgrade."
+                        $needsUpgrade = $true
+                    }
+                    else {
+                        Write-Host "::WARNING::$msg Set DependencyInstallMode to 'upgrade' to upgrade dependencies."
+                    }
+                }
+                elseif ($newVersion -lt $installedVersion) {
+                    Write-Host "::WARNING::Dependency app $($appJson.name) is already installed in version $installedVersion, which is higher than $newVersion, used for this build. Please update your local copy of this dependency."
+                }
+                else {
+                    Write-Host "Dependency app $($appJson.name) is already installed in version $installedVersion."
+                }
+            }
+            else {
+                Write-Host "Dependency app $($appJson.name) is not installed."
+                $needsInstall = $true
+            }
+            if ($needsUpgrade) {
+                if (-not $isPTE -and $installedApp.publishedAs.Trim() -eq 'Dev') {
+                    Write-Host "::WARNING::Dependency AppSource App $($appJson.name) is published in Dev scoope. Cannot upgrade."
+                    $needsUpgrade = $false
+                }
+            }
+            if ($needsUpgrade -or $needsInstall) {
+                if ($isPTE) {
+                    $PTEsToInstall += $app
+                }
+                else {
+                    Install-BcAppFromAppSource -bcAuthContext $bcAuthContext -environment $environment -appId $appJson.id -acceptIsvEula -installOrUpdateNeededDependencies
+                    # Update installed apps list as dependencies may have changed / been installed
+                    $installedApps = Get-BcInstalledExtensions -bcAuthContext $bcAuthContext -environment $environment | Where-Object { $_.isInstalled }
+                }
+            }
+        }
+        if ($PTEsToInstall) {
+            # Install or upgrade PTEs
+            Publish-PerTenantExtensionApps -bcAuthContext $bcAuthContext -environment $environment -appFiles $PTEsToInstall -SchemaSyncMode "Add"
```

Comment:
Should we use the syncMode parameter here instead of hardcoding it to "Add"?

Reply:
IMO dependencies should never be force updated - if anything we should have a DependencyInstallMode called UpgradeForce or like

Reply:
will add this